fundamentals/arrays/lesson_notes/lesson_03.md

```diff
@@ -0,0 +1,181 @@
+
+  > Your answer…
+
+- Take a moment to describe the difference between the first and second argument of `.slice()`, in your own words.
+
+  > Your answer…
+
+- Is `.slice()` a destructive or non-destructive array method? How do you know?
+
+  > Your answer…
+
+- The `.splice()` method is very similar to `.slice()`. Update your example so that it looks like the code below.
+
+  ```js
+  let coffeeAttributes = ["grapes", "tea", "chocolate"];
+  let result = coffeeAttributes.splice();
+  console.log("Coffee Attributes:", coffeeAttributes);
+  console.log("Result:", result);
+  ```
+
+  With no arguments, does `.splice()` work any differently than `.slice()`? If it does, how so?
+
+  > Your answer…
+
+- In the example above, add an argument of `1` to the `.splice()` method, like so:
+
+  ```js
+  let result = coffeeAttributes.splice(1);
+  ```
+
+  With a single argument, does `.splice()` work any differently than `.slice()`? If it does, how so?
+
+  > Your answer…
+
+- In the example above, update the arguments of `.splice()` so that it looks like the following:
+
+  ```js
+  let result = coffeeAttributes.splice(0, 2);
+  ```
+
+  With two arguments, does `.splice()` work any differently than `.slice()`? If it does, how so?
+
+  > Your answer…
+
+- In the example above, update the arguments of `.splice()` so that it looks like the following:
+
+  ```js
+  let result = coffeeAttributes.splice(1, 2, "hazelnut", "jasmine");
+  ```
+
+  After running the code above, you may be surprised at the result. Take a moment to experiment with the code to better understand how `.splice()` works. Then, in your own words, explain how the different ways the `.splice()` method works.
+
+  > Your answer…
+
+- Is `.splice()` a destructive or non-destructive array method? How do you know?
+
+  > Your answer…
+
+- The `.split()` method is not called on arrays but is instead called on strings. Run the code below and see if you can figure out how it works.
+
+  ```js
+  let coffeeAttributes = "grapes, tea, chocolate";
+  let result = coffeeAttributes.split(", ");
+  console.log("Coffee Attributes:", coffeeAttributes);
+  console.log("Result:", result);
+  ```
+
+  What data type does the `.split()` command return? How would you describe the purpose of the argument that is passed into the `.split()` method?
+
+  > Your answer…
+
+- Take a look at the following code. Before running it, mentally evaluate the code and anticipate what you think will happen.
+
+  ```js
+  let coffeeAttributes = "grapes, tea, chocolate";
+  let result = coffeeAttributes.split(", ").splice(0, 1, "hazelnut");
+  console.log("Result:", result);
+  ```
+
+  Did it work as expected? Why or why not?
+
+  > Your answer…
+
+- The `.join()` method works similar to the `.split()` method, except it can be called on arrays. Run the example below to see how it works.
+
+  ```js
+  let coffeeAttributes = ["raspberry", "floral", "honey"];
+  let result = coffeeAttributes.join(", ");
```

Comment:
This is very nuanced. Perhaps show them both `.join(",")` and  `.join(", ")` - it's very common for students to forget to put an argument for `.join("")` - another rep may be very beneficial here